spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1701,9 +1703,21 @@ def get_current_shellwidget(self) -> ShellWidget | None:
     @Slot(bool, str, str)
     @Slot(bool, bool)
     @Slot(bool, str, bool)
-    def create_new_client(self, give_focus=True, filename='', special=None,
-                          given_name=None, cache=True, initial_cwd=None,
-                          path_to_custom_interpreter=None):
+    def create_new_client(self, *args, **kwargs):
+        future = AsyncDispatcher(get_user_environment_variables)
```

Comment:
@hlouzada, this is not correct. Is there a convenient way to return a concurrent future from an awaitable future? All I can think of is to use a dummy function:
```suggestion
        @AsyncDispatcher()
        async def dummy():
            return await get_user_environment_variables()

        future = dummy()
```

Reply:
> manually wrap the future to as an awaitable inside the other `async def` with:
> 
> ```python
> awaitable = asyncio.wrap_future(
>     future, loop=asyncio.get_running_loop(),
> )
> ```

I don't understand what this is or how to apply it. Can you explain further? How would you modify `get_user_env`?

Reply:
Are you suggesting:
```python
async def get_user_env():
    """Return current user environment variables with parsed values"""
    env_dict = await asyncio.wrap_future(get_user_environment_variables(), loop=asyncio.get_running_loop())
    return envdict2listdict(env_dict)
```

Reply:
Would there be an advantage to creating both types of `get_user_environment_variables`? For example:
```python
def get_user_environment_variables():
    # ... long process
    return env_var


@AsyncDispatcher(loop="get_user_env", return_awaitable=True)
async def await_get_user_environment_variables():
    return get_user_environment_variables()


@AsyncDispatcher(loop="get_user_env")
async def future_get_user_environment_variables():
    return get_user_environment_variables()

async def get_user_env():
    """Return current user environment variables with parsed values"""
    env_dict = await await_get_user_environment_variables()
    return envdict2listdict(env_dict)
```